### DISCUSSION TITLE
How to create sections with linear distribution type

### DISCUSSION THREAD

**Discussion by lekuaa** [Dlubal-Software/RFEM_Python_Client#386] [Q&A] [answered]
Hi,

I am trying to learn this awesome Python client, but got stuck on creating members.
I can easily create a beam member with a uniform type section. But I also want to create some other distribution types, such as Linear type or Tapered type at the start of the member.
I tried the following codes to create a linear distribution type section, but it always results in a uniform type again, could anybody tell me where I did wrong?
Thank you in advance,

#----------------------------------------------------------#
# Create Nodes
#----------------------------------------------------------#
Node(1,0.0,0.0,0.0)
Node(2,5.0,0.0,0.0)
Node(3,5.0,0.0,-7.0)
Node(4,0.0,0.0,-7.0)

#----------------------------------------------------------#
# Create Material
#----------------------------------------------------------#
Material(1,'Q235')

#----------------------------------------------------------#
# Create Section
#----------------------------------------------------------#
Section(1, 'RHS 200x100x8 | GB/T 6728-2017 | --',1)
Section(2, 'RHS 400x200x10 | GB/T 6728-2017 | --',1)

#----------------------------------------------------------#
# Create Beam
#----------------------------------------------------------#
Member(1,1,4,0.0,1,2)

**Comment by heetrojivadiya** [comment, marked as answer]
Hello @lekuaa, 

Thank you for using our webservice. You can create different section distribution type beam with help of the `Member.Beam()`static method of the `Member` class. I have attached the small example so you can know how to use it.

```py
Model(True, 'demo')

Node(1,0.0,0.0,0.0)
Node(2,10.0,0.0,0.0)
Node(3,10.0,0.0,-7.0)
Node(4,0.0,0.0,-7.0)

Material(1,'Q235')

Section(1, 'RHS 200x100x8 | GB/T 6728-2017 | --',1)
Section(2, 'RHS 400x200x10 | GB/T 6728-2017 | --',1)

# Linear distribution
Member.Beam(1, 1, 2, MemberSectionDistributionType.SECTION_DISTRIBUTION_TYPE_LINEAR, start_section_no=2, end_section_no=1, distribution_parameters=[MemberSectionAlignment.SECTION_ALIGNMENT_TOP])

# Tapered at the start of member distribution
Member.Beam(2, 4, 3, MemberSectionDistributionType.SECTION_DISTRIBUTION_TYPE_TAPERED_AT_START_OF_MEMBER, start_section_no=2, end_section_no=1, distribution_parameters=[True, 0.25, MemberSectionAlignment.SECTION_ALIGNMENT_CENTRIC])

```

For creating other types of member or more details you may visit our [documentation pages](https://dlubal-software.github.io/.github/guide/RFEM.BasicObjects.html#member).

If you have any other questions then feel free to reach us.

![image](https://github.com/dlubal-software/RFEM_Python_Client/assets/108512789/cecc1ce8-fb9e-4e9c-9201-98a0383bd8c0)

**Reply by lekuaa** [reply]
Awesome! Thanks a lot for the fast response!

**Reply by lekuaa** [reply]
Hi there again, 

I am now trying to use the function Member.EndModifications() and Member.Eccentricities() which is given on the documentation pages, however, it seems these two functions have not been defined in the Member class, therefore can not be used, am I wrong? Or do I have to wait for further updates?

**Reply by heetrojivadiya** [reply]
Hello, I guess you are referring wrong documentation pages. There are no such methods in RFEM Python HLF for member. These methods are in javaScript. For correct documentation, you can follow [RFEM Python HLF](https://dlubal-software.github.io/.github/guide/ready.html) pages.

**Reply by lekuaa** [reply]
That's true, these functions are given for JavaScript. Thank you for the answer.
